js/flatbuffers.js

```diff
@@ -830,6 +839,49 @@ flatbuffers.Builder.prototype.createString = function(s) {
 flatbuffers.Builder.prototype.createLong = function(low, high) {
   return flatbuffers.Long.create(low, high);
 };
+
+/**
+ * A helper function to pack an object
+ * 
+ * @returns offset of obj
+ */
+flatbuffers.Builder.prototype.createObjectOffset = function(obj) {
+  if(obj === null) {
+    return 0
+  }
+
+  if(typeof obj === 'string') {
+    return this.createString(obj);
+  } else {
+    return obj.pack(this);
+  }
+}
+
+/**
+ * A helper function to pack a list of object
+ * 
+ * @returns list of offsets of each non null object
+ */
+flatbuffers.Builder.prototype.createObjectOffsetList = function(list) {
+  let ret = [];
+
+  for(let i = 0; i < list.length; ++i) {
+    let val = list[i];
+
+    if(val !== null) {
```

Comment:
I think this won't be an error because, in my [type file](https://github.com/khoitd1997/DefinitelyTyped/blob/patch-1/types/flatbuffers/index.d.ts), the member of the list can be null as well.